hrm-domain/hrm-core/case/caseSection/caseSectionService.ts

```diff
@@ -67,15 +108,24 @@ export const createCaseSection = async (
     createdAt: nowISO,
     accountSid,
   };
-
-  const created = await create()(record);
+  const createdResult = await create()(record);
+  if (isErr(createdResult)) {
+    if (
+      isDatabaseUniqueConstraintViolationErrorResult(createdResult) &&
+      createdResult.table === 'CaseSections'
+    ) {
+      const resourceIdentifier = `${caseId}/${sectionType}/${record.sectionId}`;
+      const cause = createdResult;
+      return newResourceAlreadyExistsResult(resourceIdentifier, cause);
+    }
+  }
```

Comment:
Shouldn't we return the error if the nested condition is not matched? :thinking:

Reply:
Yes, good catch, thx